controllers/csam-report-controller.js

```diff
@@ -0,0 +1,75 @@
+const createError = require('http-errors');
+const Sequelize = require('sequelize');
+
+const { Op } = Sequelize;
+
+const CSAMReportController = CSAMReport => {
+  /**
+   * @param {number} contactId
+   * @param {string} accountSid
+   */
+  const getCSAMReports = async (contactId, accountSid) => {
+    const queryObject = {
+      where: {
+        [Op.and]: [accountSid && { accountSid }, contactId && { contactId }],
+      },
+    };
+
+    return CSAMReport.findAll(queryObject);
+  };
+
+  /**
+   * @param {number} id
+   * @param {string} accountSid
+   */
+  const getCSAMReport = async (id, accountSid) => {
+    const queryObject = { where: { [Op.and]: [{ id }, { accountSid }] } };
+    const report = await CSAMReport.findOne(queryObject);
+
+    if (!report) {
+      const errorMessage = `CSAM Report with id ${id} not found`;
+      throw createError(404, errorMessage);
+    }
+
+    return report;
+  };
+
+  /**
+   * @param {{ twilioWorkerId: string, contactId: number, csamReportId: string }} body
+   * @param {string} accountSid
+   */
+  const createCSAMReport = async (body, accountSid) => {
+    const record = {
+      accountSid: accountSid || '',
+      twilioWorkerId: body.twilioWorkerId || '',
+      csamReportId: body.csamReportId || '',
+      contactId: body.contactId || null,
+    };
+
+    const csamReport = await CSAMReport.create(record);
```

Comment:
One general note on all of our functions that have database queries -- what happens if the query fails? What will we return?

Reply:
The endpoint returns a 500 error if I'm not wrong. Should we do an attempt to handle this kind of operations?

Reply:
Returning a 500 is reasonable for now.